### PR TITLE
refraction_angle() now accepts inc. angle

### PR DESCRIPTION
**Pull request by Trave11er**
#### Release Notes

* physics.optics
   * modified two functions in `physics.optics.utils`, namely: `refraction_angle()` and `deviation()` so they now also accept an angle of incidence. This is useful if the user wants to work the angles instead of inputting ray-like objects.

### CODE REVIEW

**Comment by sidhantnagpal** [comment]
Can you resolve the conflicts? The tests won't begin until you do that.

**Comment by oscarbenjamin** [comment]
Looks like a straight-forward conflict (click the "resolve conflicts" button below). On master a couple of new tests were added on the same lines where you've added tests. Just edit so that all tests are there together and it should be fine. I think you can do this in the browser.

**Comment by Trave11er** [comment]
Thank you, @sidhantnagpal and @oscarbenjamin, for reviewing and providing tips on how to resolve the conflict. Have done that now - should be fine.

**Comment by oscarbenjamin** [comment]
Looks like you've sorted it correctly. The tests are running now...

**Comment by Trave11er** [comment]
@oscarbenjamin please let me know if it needs more work on test coverage otherwise looks ripe to merge.

File: sympy/physics/optics/utils.py [hunk 1]
```diff
@@ -28,18 +28,31 @@
...
 from sympy.core.compatibility import is_sequence
 from sympy.geometry.line import Ray3D, Point3D
 from sympy.geometry.util import intersection
 from sympy.geometry.plane import Plane
 from .medium import Medium
 
 
+def refractive_index_of_medium(medium):
+    """
+    Helper function that return refractive index, given a medium
```
Comment (smichr) [inline sympy/physics/optics/utils.py]: return -> returns

File: sympy/physics/optics/utils.py [hunk 2]
```diff
@@ -28,18 +28,31 @@
...
+    else:
+        n = sympify(medium)
+    return n
+
+
 def refraction_angle(incident, medium1, medium2, normal=None, plane=None):
     """
     This function calculates transmitted vector after refraction at planar
     surface. `medium1` and `medium2` can be `Medium` or any sympifiable object.
+    If incident is a number then treated as angle of incidence in which case
```
Comment (smichr) [inline sympy/physics/optics/utils.py]: incident -> `incident`
angle of incidence -> angle of incidence (in radians)

File: sympy/physics/optics/utils.py [hunk 3]
```diff
@@ -92,8 +107,39 @@ def refraction_angle(incident, medium1, medium2, normal=None, plane=None):
     [-sqrt(3)*sqrt(-2*n1**2/(3*n2**2) + 1)]])
     >>> refraction_angle(r1, n1, n2, plane=P)
     Ray3D(Point3D(0, 0, 0), Point3D(n1/n2, n1/n2, -sqrt(3)*sqrt(-2*n1**2/(3*n2**2) + 1)))
-
+    >>> round(refraction_angle(0.1, 1.2, 1.5), 5)
```
Comment (smichr) [inline sympy/physics/optics/utils.py]: perhaps that 0.1 could be changed to `rad(6)` to emphasize that the input is in radians
```python
Ray3D(Point3D(0, 0, 0), Point3D(n1/n2, n1/n2, -sqrt(3)*sqrt(-2*n1**2/(3*n2**2) + 1)))

The incident angle can also be given as a value in radians:

>>> from sympy import rad, pi
>>> round(refraction_angle(pi/30, 1.2, 1.5), 5)
0.08372
>>> round(refraction_angle(rad(6), 1.2, 1.5), 5)
0.08372
```

File: sympy/physics/optics/utils.py [hunk 4]
```diff
@@ -222,16 +256,11 @@ def fresnel_coefficients(angle_of_incidence, medium1, medium2):
 
     https://en.wikipedia.org/wiki/Fresnel_equations
     """
+    if not 0.0 <= angle_of_incidence < pi*0.5:
```
Comment (smichr) [inline sympy/physics/optics/utils.py]: or `0 <= 2*angle_of_incidence < pi`

File: sympy/physics/optics/utils.py [hunk 5]
```diff
@@ -298,13 +329,22 @@ def deviation(incident, medium1, medium2, normal=None, plane=None):
...
+    >>> round(deviation(0.1, 1.2, 1.5), 5)
+    -0.02005
     """
     refracted = refraction_angle(incident,
                                  medium1,
                                  medium2,
                                  normal=normal,
                                  plane=plane)
+    try:
+        angle_of_incidence = float(incident)
```
Comment (smichr) [inline sympy/physics/optics/utils.py]: Should you be using Float instead and returning the same are does the whole optics module default to floats when using literal input?

**Comment by Trave11er** [comment]
Thank you, @smichr for reviewing - have now incorporated all your suggestions.

**Pull request merged**